app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentPresenterTest.kt

```diff
@@ -0,0 +1,342 @@
+package fr.free.nrw.commons.nearby
+
+import com.nhaarman.mockitokotlin2.*
+import fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao
+import fr.free.nrw.commons.location.LatLng
+import fr.free.nrw.commons.location.LocationServiceManager
+import fr.free.nrw.commons.location.LocationServiceManager.LocationChangeType
+import fr.free.nrw.commons.nearby.contract.NearbyParentFragmentContract
+import fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+
+/**
+ * The unit test class for NearbyParentFragmentPresenter
+ */
+class NearbyParentFragmentPresenterTest {
+    @Mock
+    internal lateinit var nearbyParentFragmentView: NearbyParentFragmentContract.View
+    @Mock
+    internal lateinit var bookmarkLocationsDao: BookmarkLocationsDao
+    @Mock
+    internal lateinit var latestLocation: LatLng
+    @Mock
+    internal lateinit var cameraTarget: LatLng
+    @Mock
+    internal lateinit var selectedLabels: List<Label>
+
+    private lateinit var nearbyPresenter: NearbyParentFragmentPresenter
+    private lateinit var mapboxCameraTarget: com.mapbox.mapboxsdk.geometry.LatLng
+
+    /**
+     * initial setup
+     */
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        nearbyPresenter = NearbyParentFragmentPresenter(bookmarkLocationsDao)
+        nearbyPresenter.attachView(nearbyParentFragmentView)
+    }
+
+    /**
+     * Tests nearby operations are initialized
+     */
+    @Test
+    fun testInitializeNearbyMapOperations() {
+        nearbyPresenter.initializeMapOperations()
+        verify(nearbyParentFragmentView).enableFABRecenter()
+        testPlacesPopulatedForLatestLocationWhenLocationSignificantlyChanged()
+        verify(nearbyParentFragmentView).addSearchThisAreaButtonAction()
+        verify(nearbyParentFragmentView).setCheckBoxAction()
+    }
+
+    /**
+     * Test lockUnlockNearby method to lock nearby case
+     */
+    @Test
+    fun testLockUnlockNearbyForLocked() {
+        nearbyPresenter.lockUnlockNearby(true)
+        verify(nearbyParentFragmentView).disableFABRecenter()
+    }
+
+    /**
+     * Test lockUnlockNearby method to unlock nearby case
+     */
+    @Test
+    fun testLockUnlockNearbyForUnlocked() {
+        nearbyPresenter.lockUnlockNearby(false)
+        verify(nearbyParentFragmentView).enableFABRecenter()
+    }
+
+    /**
+     * Test updateMapAndList method returns with zero interactions when location is locked
+     */
+    @Test
+    fun testUpdateMapAndListWhenLocationLocked() {
```

Comment:
I would rename all these test title to be in backticks so they can be an english sentence eg `When nearby is locked do not update view` or `updateMapAndList when locked does nothing` just something that expresses it well so that if somebody reads this in a failed test report they will instantly know what went wrong

Reply:
Here again I decided to rely on our previous test namings.

Reply:
The previous conversation was about documentation. I guess we should have another one on test naming style.

I personally think it is a fantastic feature that we can write test names in backticks eg
```
fun `given some state produce this result`(){}
```
as opposed to

```
fun givenSomeStateProduceThisResult(){}
```

@misaochan @ashishkumar468 @maskaravivek

Reply:
Although the backticks style does seem reasonable, but before we impose anything like that, we should discuss and agree to it. Contributors who are willing to use the same are more than welcome, but before we have all agreed to impose it, IMO we can accept PR's who use the existing naming conventions

Reply:
We didn't all unanimously agree to use 
````
whenever
````
 instead of
```
Mockito.`when`
```

but it looks a lot better so everyone did. We didn't have to update a bible or mark all previous usages as wrong, just commit to doing the better thing going forward which is a perfectly fine and organic method of disseminating information to me.

If needs must I will create an issue.

Reply:
Yes, please create an issue, so that we can discuss it and update the docs if we all agree. :) If everything else is OK with this PR we can merge, renaming could  be a good first issue for a new contributor later on, no need to force it now.

Reply:
There is nothing to update as there is no test style document I saw
Anyway #3638 created